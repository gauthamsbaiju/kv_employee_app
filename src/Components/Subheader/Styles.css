.subheader{
    width: 100%;
height: 100px;
background: #FDFDFD 0% 0% no-repeat padding-box;
box-shadow: 0px 3px 6px #9D9D9D29;
border-radius: 10px;
opacity: 1;
display: flex;
align-items: center;
justify-content: space-between;
flex-wrap: nowrap;
padding-right: 20px;
}

.text{

text-align: left;
font: normal normal bold 24px/32px Nunito Sans;
letter-spacing: 0px;
color: #000000;
opacity: 1;
padding-left: 24px;
}

.filter-text{
/* margin-left: 600px ; */
padding-right: 20px;

text-align: left;
font: normal normal normal 18px/24px Nunito Sans;
letter-spacing: 0px;
color: black;
opacity: 1;
}

.filter{
    top: 126px;
left: 1389px;
width: 161px;
height: 31px;
margin-right: 68px;

background: #EAF9FF 0% 0% no-repeat padding-box;
border-radius: 50px;
opacity: 1;
}


.empbtn{
    width: 259px;
    /* width:25%; */
    height: 58px;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-top: 0px;
    
    text-align: left;
    font: normal normal normal 20px/27px Nunito Sans;
    letter-spacing: 0px;
    background-color: #EAF9FF;
    border-radius: 50px;
    border: 0px;
}

.butn-text{
    color: black;
    width: 200px;
    padding-left: 10px;
    text-align: left;
font: normal normal normal 20px Nunito Sans;
}

.rightelem{
    display: flex;
    flex-direction: row;
    align-items:center;
    justify-content:space-between
}

.circle{
    height: 58px;
    width: 58px;
    border-radius: 50%;
    background-color: #03AEEE;
    text-align: center;
    color: white;
    border: 0px;
    font-size: 35px;
    padding-bottom: 5px;
}
