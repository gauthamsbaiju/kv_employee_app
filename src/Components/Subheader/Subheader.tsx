import React from 'react';
import './Styles.css';
import { useNavigate, useParams } from 'react-router-dom';

export type SubheaderPropsType = {
  title: string;
  filter: boolean;
  button: boolean;
  text: string;
  type: string;
};

const Subheader: React.FC<SubheaderPropsType> = (props) => {
  const navigate = useNavigate();
  const { id } = useParams();
  const handleClick = () => {
    if (props.type === 'Create') navigate(`/employee/create_employee`);
  };
  const handleClick2 = () => {
    if (props.type === 'Edit') navigate(`/employee/edit/${id}`);
  };
  const role = localStorage.getItem('Role');

  return (
    <div className='subheader-wrapper' data-testid='subheader-test'>
      <div className='subheader'>
        <div className='text' data-testid='subheaderTitle-test'>
          {props.title}
        </div>
        <div className='rightelem' data-testid='subheaderFilter-test'>
          {props.filter && (
            <>
              <div className='filter-text'>Filter By</div>
              <div className='filter'></div>
            </>
          )}

          {role === 'admin' && props.button && props.type !== 'Edit' && (
            <div className='empbtn'>
              <button
                className='circle'
                onClick={() => {
                  handleClick();
                }}
              >
                +
              </button>
              <div className='butn-text'>{props.text}</div>
            </div>
          )}
          {role === 'admin' && props.button && props.type === 'Edit' && (
            <div className='empbtn'>
              <button
                className='circle'
                onClick={() => {
                  handleClick2();
                }}
              >
                <img src='/assets/icons/Path 327.svg' alt='e'></img>
              </button>
              <div className='butn-text' data-testid='subheaderText-test'>
                {props.text}
              </div>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

export default Subheader;
