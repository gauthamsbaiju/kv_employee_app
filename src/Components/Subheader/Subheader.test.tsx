import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Subheader, { SubheaderPropsType } from './Subheader';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate
}));

describe('Test for subheader component', () => {
  test('Test for title rendered', () => {
    const SubheaderProps: SubheaderPropsType = {
      title: 'Subheader',
      filter: true,
      button: true,
      text: 'Subheader',
      type: 'Edit'
    };

    render(<Subheader {...SubheaderProps} />);
    const element = screen.getByTestId('subheaderTitle-test');

    if (element) expect(element).toHaveTextContent('Subheader');
  });

  //   test('Test for text rendered', () => {
  //     const SubheaderProps: SubheaderPropsType = {
  //       title: 'Subheader',
  //       filter: true,
  //       button: true,
  //       text: 'Subheader',
  //       type: 'Edit'
  //     };

  //     render(<Subheader {...SubheaderProps} />);
  //     const element = screen.getByTestId('subheaderText-test');

  //     if (element) expect(element).toHaveTextContent('Subheader');
  //   });

  //   test('Test for input type', () => {
  //     const SubheaderProps: SubheaderPropsType = {
  //       title: 'Subheader',
  //       filter: true,
  //       button: true,
  //       text: 'Subheader',
  //       type: 'Edit'
  //     };

  //     render(<Subheader {...SubheaderProps} />);
  //     const element = screen.getByTestId('subheaderFilter-test');

  //     expect(element.children).toHaveAttribute('filter', true);
  //   });

  //   test('Test for input type', () => {
  //     const SubheaderProps: SubheaderPropsType = {
  //       title: 'Subheader',
  //       filter: true,
  //       button: true,
  //       text: 'Subheader',
  //       type: 'Edit'
  //     };

  //     render(<Subheader {...SubheaderProps} />);
  //     const element = screen.getByTestId('subheaderFilter-test');

  //     expect(element).toHaveAttribute('button', true);
  //   });

  //   test('Test for input placeholder', () => {
  //     const InputProps: InputPropsType = {
  //       lable: 'Input',
  //       placeholder: 'Input',
  //       type: 'text'
  //     };

  //     render(<Input {...InputProps} />);
  //     const element = screen.getByTestId('input-test');

  //     expect(element).toHaveAttribute('placeholder', 'Input');
  //   });

  //   test('Test for on change', () => {
  //     const onChange = jest.fn();
  //     const InputProps: InputPropsType = {
  //       lable: 'Input',
  //       placeholder: 'Input',
  //       type: 'text',
  //       onChange
  //     };

  //     render(<Input {...InputProps} />);
  //     const element = screen.getByTestId('input-test');

  //     fireEvent.change(element);
  //     expect(onChange).toBeCalled;
  //   });

  test('Test for snapshot rendered', () => {
    const SubheaderProps: SubheaderPropsType = {
      title: 'Subheader',
      filter: true,
      button: true,
      text: 'Subheader',
      type: 'Edit'
    };

    render(<Subheader {...SubheaderProps} />);
    const element = screen.getByTestId('subheader-test');

    expect(element).toMatchSnapshot();
  });
});
