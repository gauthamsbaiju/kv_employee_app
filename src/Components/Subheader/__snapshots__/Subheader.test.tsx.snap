// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for subheader component Test for snapshot rendered 1`] = `
<div
  class="subheader-wrapper"
  data-testid="subheader-test"
>
  <div
    class="subheader"
  >
    <div
      class="text"
      data-testid="subheaderTitle-test"
    >
      Subheader
    </div>
    <div
      class="rightelem"
      data-testid="subheaderFilter-test"
    >
      <div
        class="filter-text"
      >
        Filter By
      </div>
      <div
        class="filter"
      />
    </div>
  </div>
</div>
`;
